src/EditorFeatures/Core/Shared/Utilities/VisualStudioTaskSchedulerProvider.cs

```diff
@@ -2,28 +2,30 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System;
 using System.Collections.Generic;
 using System.Composition;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.VisualStudio.Threading;
+using Roslyn.Utilities;
 
-namespace Microsoft.VisualStudio.LanguageServices.Implementation
+namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
 {
-    [ExportWorkspaceService(typeof(ITaskSchedulerProvider), ServiceLayer.Host), Shared]
+    [ExportWorkspaceService(typeof(ITaskSchedulerProvider), ServiceLayer.Editor), Shared]
```

Comment:
I admit I wish we had a comment here that this is the default being provided for back compat, but we encourage new hosts who can do so to override this behavior and run WorkspaceChanged off the main thread.

Reply:
If new hosts are out-of-process, they wouldn't have a main thread and this is automatic. 😄

Reply:
Well, out of proc hosts aren't importing this layer in the first place. I was thinking more for things like VS for Mac or something. But yeah, not sure how many other ones we might still have of that.